### DISCUSSION TITLE
Is there grounds to build a unifying model between trust registries and reputation systems?

### DISCUSSION THREAD

**Discussion by andorsk** [trustoverip/tswg-trust-registry-tf#75] [General]
I'm compiling a [document here](https://hackmd.io/@andorsk/HkRzxYsas) that discusses this in more detail ( it will take me some time to create so there's not much there at the time of this posting ), but the question I want ask is whether there's a unified model to describe both reputation systems and registry systems. 

There are some unique features about a registry vs. a reputation system but I'd argue there's probably more similarities than differences. 

On the differences, they are more numerous, but I'd argue less fundamental and often allow exceptions: Reputation systems tend to be distributed. They tend to leverage the community for improving the trust network. They tend to be less "official" as they are not generally authorized by a legislative body. Meanwhile, registry systems tend to be run by a governance authority, and they tend to be built as a "list".  They often have different technical requirements at an implementation level. 

On the similarities: They are both used to establish credibility within a trust decision, and both rely on context to be relevant. They both have governance frameworks. These are critical and fundamental attributes of reputation and registry frameworks.

My question is two fold: 

- Are the high level interfaces the same between a trust registry and a reputation system? If they are different, how and why? 
- Is there a fundamental model that describes both systems and how can we leverage that model to build better technology? 

I am bootstrapping this question by putting this in Discussions a little premature, to see if anyone wishes to expand and assist in building the mental model around these two frameworks. I will expand on my thoughts with a literature review and background when I have the time to get a proper document together collating my thoughts. 

If anyone wants to contribute or help, I would love to collaborate.

**Comment by a-fox** [comment]
I think this is most of all a question to consider from the relying party perspective. Especially I would think back from the question _“What does the relying party actually base their trust on?”_

**In Trust Registries** it’s fairly straightforward:
1) An entity signs a contract/agreement to be part of an ecosystem
2) contract stipulates compliancy requirements
3) contract gives eecosystem governance authority the power to maintain compliancy
4) governance authority has strong capability to ensure compliancy and can vouch for its ecosystem members because of the ecosystem audit results
Ergo: relying party trusts the governance authority’s power to guarantee compliancy

**In reputation systems** it’s not that simple:
1) A governance entity can provide a common evaluation schema, and perhaps even a registry to store the evaluations (unless everybody holds their own reputation results).
2) The governance entity can only vouch for the schema and the registry, nothing else.
3) In a FAITHful system (look at me, using already Daniel’s cute wordings), we can guarantee the authenticity of the message, but we can NEVER guarantee if the evaluation was truthful, i.e. if the vote was done with malicious intent.
Ergo: relying party trusts governance entity & the FAITHful system. But response can never be binary and guaranteed, but on a scale.

![Screenshot 2023-02-16 at 22 42 30](https://user-images.githubusercontent.com/1370528/219483420-74e2e17f-7571-41af-a7a1-cbc28432fb74.jpeg)
(sorry crude image)

**To summarise:**
- Registry mechanism is deterministic, and can be binary or anything else that fits the governance scheme
- Trust Reputation mechanism is always probabilistic and likely on a scale.

**The question of interfaces** comes down to the system used to store & implement reputation inputs. If it's a registry-type of system (e.g. blockchain to store all evaluations), it's just a simple Trust Registry query.
If it would be maintained by the evaluated entity themselves (e.g. on a DWN, wallet, etc), then  we would need a corresponding and interoperable trust task on the entity's wallet that can provide thee responce and evidence of immutability and authenticity of its data.

IMO the context question is the same in both. It's a matter of parametrizing the different query types & expected input.
